zfsbootmenu/bin/reboot

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# vim: softtabstop=2 shiftwidth=2 expandtab
+
+# shellcheck disable=SC1091
+sources=(
+  /lib/zfsbootmenu-core.sh
+)
+
+for src in "${sources[@]}"; do
+  # shellcheck disable=SC1090
+  if ! source "${src}" >/dev/null 2>&1 ; then
+    echo -e "\033[0;31mWARNING: ${src} was not sourced; unable to proceed\033[0m"
+    exit 1
+  fi
+done
+
+unset src sources
+
+while read -r _pool; do
+  is_writable "${_pool}" && export_pool "${_pool}"
+done <<<"$( zpool list -H -o name )"
+
+[ -w /proc/sysrq-trigger ] && echo b > /proc/sysrq-trigger
```

Comment:
If this file doesn't exist or isn't writable, this should print a message instructing the user to hard-reset the system because a soft reboot is not supported.

Reply:
Good idea!